microservices.yml

```diff
@@ -58,16 +59,36 @@ commands:
           name: Set Full Image Name
           command: |
             echo 'export TAG_LATEST="${ECR_APP_IMAGE}:latest"' >> $BASH_ENV
-            echo 'export TAG_CI="${ECR_APP_IMAGE}:BUILD_${CIRCLE_BUILD_NUM}"' >> $BASH_ENV
+            echo 'export TAG_BRANCH="${ECR_APP_IMAGE}:BRANCH_${CIRCLE_BRANCH/\//_}"' >> $BASH_ENV
             echo 'export TAG_COMMIT="${ECR_APP_IMAGE}:COMMIT_${CIRCLE_SHA1}"' >> $BASH_ENV
+            [ ! -z $CIRCLE_TAG ] && echo 'export TAG_GITTAG="${ECR_APP_IMAGE}:${CIRCLE_TAG}"' >> $BASH_ENV || true
 
   build-and-tag-docker-image:
     description: Builds the docker image and tags it with commit and branch
     steps:
       - run:
           name: Build Docker Image
           command: |
-            docker build -t $TAG_LATEST -t $TAG_CI -t $TAG_COMMIT -f Dockerfile .
+            docker build -t $TAG_LATEST -t $TAG_BRANCH -t $TAG_COMMIT -f Dockerfile .
+            [ ! -z $TAG_GITTAG ] && docker tag $TAG_COMMIT $TAG_GITTAG || true
+      - run:
+          name: Save docker image
```

Comment:
I don't think this matters since it is a VM but do we need a "cleanup" task to delete what is in the images directory?

Reply:
afaik the VMs that are used in CircleCI get trashed after use (you can "restart tests with ssh" and then VM is kept for up to 2h)

sooo i wouldn't bother with cleanup